### DISCUSSION TITLE
How to trigger smart contract properly

### DISCUSSION THREAD

**Discussion by SembaMax** [trustwallet/wallet-core#1654] [Q&A]
Hello,

I want to ask how to trigger smart contract method on Tron transaction.

this's how far i reached, but still cannot figure out how to pass the function name or function's parameters.

`val transferContract = Tron.TriggerSmartContract.newBuilder()
            .setOwnerAddress(tronAddress)
            .setCallValue(0)
            .setContractAddress(contractAddress)
            .build()`

and here's a js example of what i'm trying to implement:

`tronWeb.transactionBuilder.triggerSmartContract(
        CONTRACT, 'transfer(address,uint256)', {
            feeLimit: 1_000_000,
            callValue: 0
        },
        [{
            type: 'address',
            value: ACCOUNT
        }, {
            type: 'uint256',
            value: 1000000
        }]
    );`

Thanks in advance!

**Comment by optout21** [comment]
Have you checked Tron signing tests?
https://github.com/trustwallet/wallet-core/blob/master/tests/Tron/SignerTests.cpp

**Comment by SembaMax** [comment]
@catenocrypt yes, i check this test file.
but i still cannot see any explicit way to provide the "function_selector" and its parameters on smart contract.

for example i need to pass:
function_selector: commitAmount(uint256)
parameter: {type:uint256, value:amount}

**Comment by SembaMax** [comment]
can i define the "function_selector" using "set_data()" ?
if so, i would be grateful if you can explain how to use or refer to an example.
